apps/assessments/models.py

```diff
@@ -0,0 +1,170 @@
+# Django Imports
+from django.conf import settings
+from django.db import models
+
+# HTK Imports
+from htk.utils import htk_setting
+
+
+class AbstractAssessment(models.Model):
+    """Main class for Assessments app.
+
+    Possible use cases include:
+    - Quizzes and Exams
+    - Surveys
+    """
+
+    name = models.CharField(max_length=255, blank=True)
+    # Tracks the version of the assessment
+    version = models.IntegerField(default=1)
+    # Number of attempts allowed: 0 for unlimited attempts, 1 for only 1.
+    num_allowed_attempts = models.PositiveIntegerField(default=0)
+    # when `True` and `num_allowed_attempts` == 1, previous responses are overwritten
+    is_repeat_allowed = models.BooleanField(default=False)
+    # Optional "Exit message" to display when an incorrect answer is provided
+    knockout_message = models.TextField(blank=True, null=True)
+
+    class Meta:
+        abstract = True
+
+    def __str__(self):
+        return f"{self.name} (Version {self.version})"
+
+
+class AbstractAssessmentQuestion(models.Model):
+    """Represents one question in an Assessment."""
+
+    ASSESSMENT_MODEL = htk_setting('HTK_ASSESSMENT_MODEL')
+
+    QUESTION_TYPES = (
+        ('FR', 'Free Response'),
+        ('MC', 'Multiple Choice'),
+        ('YN', 'Yes or No'),
+    )
```

Comment:
Can we convert this to an enum? in `apps/assessments/enums.py`
Also can we please use numbers instead of 2 character strings?

Pseudo code.
```python
class QuestionType(enum):
    UNSPECIFIED = 0
    FREE_RESPONSE = 100
    MULTIPLE_CHOICE = 200
    YES_OR_NO = 300